### PR TITLE
[IE CLDNN] USM refactoring

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
### Details:
 - Added `UsmHelper` class which loads and holds all function ptrs to work with USM
 - `UsmHelper` is created once for each `ocl_engine` instance
 - enabled cpplint for runtime component
 - used `load_entry_point` function for other extensions (surface sharing)
 -  suppressed warning from old GCC about pointer-to-object to pointer-to-function cast.

### Tickets
 - 46420
 - 28579

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/runtime/ocl/ocl_ext.hpp [hunk 1]
```diff
@@ -240,529 +157,551 @@ typedef CL_API_ENTRY cl_mem(CL_API_CALL * PFN_clCreateFromMediaSurfaceINTEL)(
...
+        std::vector<cl_mem>& surfaces,
+        cl_int * err = NULL)
+        : m_queue(queue), m_surfaces(surfaces), m_errPtr(err) {
 
-            if (pfn_acquire!=NULL && m_surfaces.size()) {
-                cl_int error = pfn_acquire(m_queue,
-                    static_cast<cl_uint>(m_surfaces.size()),
-                    m_surfaces.data(),
-                    0, NULL, NULL);
+        if (pfn_acquire!=NULL && m_surfaces.size()) {
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/runtime/ocl/ocl_ext.hpp:192]: Please add spaces around "!=".

Comment (vladimir-paramuzov) [reply inference-engine/thirdparty/clDNN/runtime/ocl/ocl_ext.hpp]: enabled cpplint for runtime and fixed all reported issues

File: inference-engine/thirdparty/clDNN/runtime/ocl/ocl_ext.hpp [hunk 2]
```diff
@@ -240,529 +157,558 @@ typedef CL_API_ENTRY cl_mem(CL_API_CALL * PFN_clCreateFromMediaSurfaceINTEL)(
...
+        } else {
+            _host_mem_alloc_fn             = nullptr;
+            _shared_mem_alloc_fn           = nullptr;
+            _device_mem_alloc_fn           = nullptr;
+            _mem_free_fn                   = nullptr;
+            _set_kernel_arg_mem_pointer_fn = nullptr;
+            _enqueue_memcpy_fn             = nullptr;
+            _enqueue_mem_fill_fn           = nullptr;
+            _enqueue_memset_fn             = nullptr;
         }
```
Comment (amolotko) [inline inference-engine/thirdparty/clDNN/runtime/ocl/ocl_ext.hpp:514]: My minor comment is to move initialization of these variables by null to in-class member initialization and remove from constructor

Comment (vladimir-paramuzov) [reply inference-engine/thirdparty/clDNN/runtime/ocl/ocl_ext.hpp:505]: Good point. fixed

**Review by amolotko — APPROVED** [review]
Looks good for me

**Pull request merged**